src/DGtal/math/MultiStatistics.h

```diff
@@ -0,0 +1,461 @@
+  
+
+
+    /** 
+     * Adds a new sample value [v] for the variable [k].
+     * 
+     * @param[in] k the statistical variable.
+     * @param[in] v the new sample value.
+     */
+    void addValue( unsigned int k, double v );
+
+  
+    /**
+     * Adds a sequence of sample values for the variable [k], scanning
+     * a container from iterators [b] to [e].
+     * 
+     * Exemple: 
+     @code
+     vector<double> x;
+     MultiStatistics stats( 1, true );
+     stats.addValue( x + 4, x + 10 );
+     @endcode
+     *
+     * @param[in] k the statistical variable.
+     * @param[in] b an iterator on the starting point.
+     * @param[in] e an iterator after the last point.
+     */
+    template <class Iter>
+    void addValues(const  unsigned int k, Iter b, Iter e );
+
+    /** 
+     * Once all sample values have been added to this object, computes
+     * meaningful statistics like sample mean, variance and unbiased
+     * variance.
+     * 
+     * @see mean, variance, unbiasedVariance, min, max
+     */
+    void terminate();
+
+
+    /** 
+     * Reinitializes the object to contain [size] statistical
+     * variables. All preceding sample values and statistics are
+     * lost. Calls 'erase()'.
+     * 
+     * @param[in] size the number of statistical variables.
+     *
+     * @param[in] storeSamples when 'true', stores each given sample,
+     *
+     * @see erase
+     */
+    void init( unsigned int size, bool storeSamples );
+
+
+    /** 
+     * Clears the object from all given samples. The object still
+     * represents 'nb()' statistical variables.
+     */
+    void clear();
+  
+
+    /** 
+     * Erases the object as if it was just created. The user must call
+     * 'init()' before any other method.
+     * 
+     * @see init
+     */
+    void erase();
+
+    /**
+     * Computes the covariance of the [x],[y] variables. The
+     * 'MultiStatistics' object should have stored its values and
+     * 'terminate' should have been called.
+     *
+     * @param[in] x the index of the x-variable (should be valid )
+     * @param[in] y the index of the y-variable (should be valid )
+     * @param[in] s the starting value index (default is 0 )
+     * @param[in] e the after-the-end value index (default is 0, meaning all )
+     *
+     * @return the covariance of [x],[y] from indices [s] to [e].
+     */
+  double covariance( const unsigned int x, const unsigned int y, 
+                     const unsigned int s = 0, unsigned int e = 0 ) const;
+  
+    /**
+     * Computes the linear regression of the ([x],[y]) variables. The
+     * 'MultiStatistics' object should have stored its values and
+     * 'terminate' should have been called.
+     *
+     * @param[in] x the index of the x-variable (should be valid )
+     * @param[in] y the index of the y-variable (should be valid )
+     *
+     * @return the pair (slope, shift at origin )
+     */
+    std::pair<double,double> linearRegression( const unsigned int x, 
```

Comment:
"x the index of the statistics to be used as the x-variable (should be valid)"  instead ? (same for y?)